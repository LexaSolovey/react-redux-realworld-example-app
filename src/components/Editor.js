import React from 'react';

export default function Editor (props) {
  const removeTagHandler = (tag) => this.props.onRemoveTag(tag);

  const watchForEnter = event => {
    if (event.keyCode === 13) {
      event.preventDefault();
      this.props.onAddTag();
    }
  }

  const onUpdateField = ({target}) => this.props.onUpdateField(target.name, target.value);

  return (
    <form>
      <fieldset>
        <fieldset className="form-group">
          <input
            className="form-control form-control-lg"
            type="text"
            name="title"
            placeholder="Article Title"
            value={props.title}
            onChange={onUpdateField} />
        </fieldset>

        <fieldset className="form-group">
          <input
            className="form-control"
            type="text"
            name="description"
            placeholder="What's this article about?"
            value={props.description}
            onChange={onUpdateField} />
        </fieldset>

        <fieldset className="form-group">
          <textarea
            className="form-control"
            rows="8"
            name="body"
            placeholder="Write your article (in markdown)"
            value={props.body}
            onChange={onUpdateField}>
          </textarea>
        </fieldset>

        <fieldset className="form-group">
          <input
            className="form-control"
            type="text"
            name="tagInput"
            placeholder="Enter tags"
            value={props.tagInput}
            onChange={onUpdateField}
            onKeyUp={watchForEnter}
          />
          <div className="tag-list">
            {(props.tagList || []).map(tag => {
                return (
                  <span className="tag-default tag-pill" key={tag}>
                    <i  className="ion-close-round"
                        onClick={() => removeTagHandler(tag)}>
                    </i>
                    {tag}
                  </span>
                );
              })
            }
          </div>
        </fieldset>

        <button
          className="btn btn-lg pull-xs-right btn-primary"
          type="button"
          disabled={props.inProgress}
          onClick={props.submitForm}
        >
          Publish Article
        </button>

      </fieldset>
    </form>
  );
}
