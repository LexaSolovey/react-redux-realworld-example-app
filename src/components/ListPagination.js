import React from 'react';

const ListPagination = (props) => {
    if (props.articlesCount <= 10) {
        return null;
    }

  const range = [];
  for (let i = 0; i < Math.ceil(props.articlesCount / 10); ++i) {
    range.push(i);
  }

  const onTagClick = (event) => {
    event.preventDefault();

    const {tagName, attributes} = event.target;
    if (tagName === 'A' && attributes['data-tagname'].value) {
      props.setPage(+ attributes['data-tagname'].value)
    }
  };

  return (
    <nav>
      <ul className="pagination" onClick={onTagClick}>
        {range.map((item) => (
          <li
            className={ item === props.currentPage ? 'page-item active' : 'page-item' }
            key={item.toString()}
          >
            <a data-tagname={item} className="page-link" href="">{item + 1}</a>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default ListPagination;
