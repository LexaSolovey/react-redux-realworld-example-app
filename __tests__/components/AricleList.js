import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { expect as chaiExpext } from 'chai';

import { articles } from '../data/articleList'; 
import { ArticleList } from '../../src/components/ArticleList';
import ArticlePreview from '../../src/components/ArticlePreview';


describe('>> ArticleList component', () => {
    it('snapshot returns Loading in case when articles is not exist', () => {
        const wrapper = renderer.create(<ArticleList />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('snapshot returns no articles', () => {
        const wrapper = renderer.create(<ArticleList articles={[]} />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
    
    it('renders correct number of items', () => {
        const wrapper = shallow(<ArticleList articles={articles} articlesCount={articles.length} />)
        chaiExpext(wrapper.find(ArticlePreview)).to.have.lengthOf(articles.length);
    });
})
