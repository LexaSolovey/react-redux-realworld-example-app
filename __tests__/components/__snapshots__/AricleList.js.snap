// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> ArticleList component snapshot returns Loading in case when articles is not exist 1`] = `
<div
  className="article-preview"
>
  Loading...
</div>
`;

exports[`>> ArticleList component snapshot returns no articles 1`] = `
<div
  className="article-preview"
>
  No articles are here... yet.
</div>
`;
